### DISCUSSION TITLE
Hide slider thumb and show it on click

### DISCUSSION THREAD

**Discussion by yyyxiang** [jspsych/jsPsych#862] [Q&A] [answered]
Hi all! I'm using html-slider-response. I want to hide the slider thumb at first (so that there's no bias), and let it appear after the participant clicks on the slider runnable track. By changing css in <style> in the head, I can now hide the thumb, but I don't know how to let it appear again. My intuition is that I can use on_load on the trial and then define a function which changes the css onclick. Can someone help with that? :)

Thanks a lot!
Yang

**Comment by jodeleeuw** [comment, marked as answer]
Hi Yang,

I think your intuition is right. I'm not sure what the CSS is to hide the thumb, but here's a general approach:

```js
on_load: function(){
	document.querySelector('#jspsych-html-slider-response-response').addEventListener('click', function(e){
		e.target.style.visibility = 'hidden'
	})
}
```

This will hide the slider when someone clicks on it, but you can adapt as needed 😄

**Reply by yyyxiang** [reply]
Thank you! :)

**Reply by SaoirseConnorDesai** [reply]
Hi there! I was wondering if you could share how you changed the CSS to hide the thumb? I tried the following in my CSS file:

```
#jspsych-html-slider-response {
    -webkit-appearance: none;
    width: 100%;
    height: 20px;
    background: #d3d3d3;
  }
```

And the on_load function above, but the thumb still appears in the centre of the slider scale.

**Reply by yyyxiang** [reply]
Hi! I didn't actually work it out finally. I found that you can set the visibility of the slider thumb to "hidden" in CSS, but the on_load function cannot modify the slider thumb only (it has to work with the entire slider). So I finally just used survey-likert for discrete responses instead.

**Reply by jodeleeuw** [reply]
Hi @yyyxiang 

Another option would be to embed a `<style>` element in the stimulus or using the `on_load` function to add the CSS rules to the page. Then you could modify that `<style>` element to change the visibility of the thumb.

**Reply by yyyxiang** [reply]
Thank you Josh! Let me try that :)
